service.py

```diff
@@ -21,7 +27,16 @@ def error(self, msg):
 
     def fail(self, msg):
         print(msg, file=stderr)
-        self.send_error(500, explain=msg)
+        # create our own error response rather than using send_error to
+        # avoid bloating response with HTML wrapping
+        response_bytes = msg.encode()
+        self.send_response(500)
+        self.send_header('Content-Type', 'text/plain')
+        self.send_header('Content-Length', len(response_bytes))
+        self.end_headers()
+        self.wfile.write(response_bytes)
```

Comment:
You can probably just do this, or move `response_bytes = msg.encode()` before the line to make it more readable, which avoids going back to check what `response_bytes` is when you read it. But this is just my preference.
```suggestion
        self.send_response(500)
        self.send_header('Content-Type', 'text/plain')
        self.send_header('Content-Length', len(response_bytes))
        self.end_headers()
        self.wfile.write(msg.encode())
```

Reply:
@mattd233, but we're using response_bytes to calculate Content-Length. Unless you are also suggesting `len(msg.encode())` ?

Reply:
Ah, didn't see that, my bad, ignore this then.